items/item-history.js

```diff
@@ -388,29 +377,32 @@ class ItemHistory {
   /**
    * @private
    */
-  _stateOrNull (result) {
-    return result === null ? null : result.getState().toString();
-  }
-
-  /**
-   * @private
-   */
-  _timestampOrNull (result) {
-    return result === null ? null : time.ZonedDateTime.parse(result.getTimestamp().toString());
+  _dateOrNull (result) {
+    return result === null ? null : time.ZonedDateTime.parse(result.toString());
   }
 
   /**
    * @private
    */
-  _dateOrNull (result) {
-    return result === null ? null : time.ZonedDateTime.parse(result.toString());
+  _decimalOrNull (result) {
+    return result === null ? null : result.toBigDecimal();
   }
 
   /**
    * @private
    */
-  _decimalOrNull (result) {
-    return result === null ? null : result.toBigDecimal();
+  _historicItemOrNull (result) {
+    if (result === null) {
+      return null;
+    }
+    const rawState = result.getState();
+    const numericState = parseFloat(rawState.toString());
+    return {
+      state: rawState.toString(),
+      rawState: rawState,
+      numericState: numericState == NaN ? null : numericState,
```

Comment:
```suggestion
      numericState: isNaN(numericState) ? null : numericState,
```

To fix the linting errors.

Reply:
Silly mistake on my part :-/ ... fixed.